### PR TITLE
Add CONTRIBUTING document

### PR DESCRIPTION
**Pull request by jbertran**
#### WHAT

Add `CONTRIBUTING.md`

#### WHY

Let's start having guidelines set in stone elsewhere than a google doc.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Contributing to the Project
+
+This document contains and defines the rules that have to be followed by any
+contributor to the project, in order for any change to be merged into the
+stable branches.
+
+## Workflow
+
+The workflow followed is waterfall. In short, a contributor will target his
```
Comment (NicolasT) [inline CONTRIBUTING.md]: `waterflow`. The `waterfall` development model is something else :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+contributor to the project, in order for any change to be merged into the
+stable branches.
+
+## Workflow
+
+The workflow followed is waterfall. In short, a contributor will target his 
+work on the lowest version branch possible starting with 2.0, and the changes
+will automatically be forward-ported to higher versions.
+
+## Restrictions of the master branch
```
Comment (NicolasT) [inline CONTRIBUTING.md]: There is no `master` branch, so this is a bit confusing.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+The master branch is a very specific branch in our workflow. No commit must
+ever go directly into the master branch, and the code shall follow one and only
+one path to get into the master branch.
+
+Merging a pull request into the master branch will only ever be done by the
+Bert-E merging bot. Once the necessary approvals and CI passes are met, one 
+only needs to comment `@Bert-E_scality approve` to trigger the merging process
```
Comment (NicolasT) [inline CONTRIBUTING.md]: Isn't it simply `@bert-e approve`?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+* relevant to the goal of the changes made
+
+### Committing Guidelines
+
+No restrictions are placed at this time on individual commits passing in the
+CI and/or maintaining full functionality of the repository.
+
+Commit messages should:
+
+* have a short (< 60 characters) summary as title
```
Comment (NicolasT) [inline CONTRIBUTING.md]: I believe the Git standard is 52

File: README.md [hunk 5]
```diff
@@ -3,3 +3,8 @@
 </h1>
 
 An opinionated Kubernetes distribution with a focus on long-term on-prem deployments
+
+## Contributing
+
+If you'd like to contribute, please review the 
+[Contributing Guidelines](CONTRIBUTING.md)
```
Comment (NicolasT) [inline README.md]: `.` at the end of a sentence.

**Review by NicolasT — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (jbertran) [reply CONTRIBUTING.md]: going on memory from bitbucket workflow doesn't work too well :upside_down_face:

File: CONTRIBUTING.md (hunk 2 above)
Comment (jbertran) [reply CONTRIBUTING.md]: Changed to `development/X.Y`

File: CONTRIBUTING.md (hunk 1 above)
Comment (jbertran) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 4 above)
Comment (slaperche-scality) [reply CONTRIBUTING.md]: Less than 50 in fact, according to the man page of `git commit`

> Though not required, it’s a good idea to begin the commit message with a single short (less than 50 character) line summarizing the change, followed by a blank line and then a more thorough description. The text up to the first blank line in a commit message is treated as the commit title, and that title is used throughout Git. For example, git-format-patch(1) turns a commit into email, and it uses the title on the Subject line and the rest of the commit in the body

The general rule is often called the "50/72 rule".

Comment (jbertran) [reply CONTRIBUTING.md]: Done

File: README.md (hunk 5 above)
Comment (jbertran) [reply README.md]: And done.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
+# Contributing to the Project
+
+This document contains and defines the rules that have to be followed by any
+contributor to the project, in order for any change to be merged into the
+stable branches.
+
+## Workflow
+
+The workflow followed is waterflow. In short, a contributor will target his
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
The workflow followed is [GitWaterFlow](https://dl.acm.org/citation.cfm?id=2993277) ([PDF](https://mt.scality.com/asset/169:releng-release-process)). In short, a contributor will target his
```

Comment (NicolasT) [inline CONTRIBUTING.md]: Let's try not to use too much `his`:
```suggestion
The workflow followed is waterflow. In short, a contributor will target changes to
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+work on the lowest version branch possible starting with 2.0, and the changes
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
the lowest version branch impacted, starting with 2.0, and the changes
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+will automatically be forward-ported to higher versions.
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
will automatically be forward-ported to more recent versions.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+
+## Restrictions of the development/X.Y branch
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
## Restrictions on the development/X.Y branches
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+The `development/X.Y` branch is a very specific branch in our workflow. No
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
The `development/X.Y` branches are very specific in the GitWaterFlow workflow. No 
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+## Workflow
+
+The workflow followed is waterflow. In short, a contributor will target his
+work on the lowest version branch possible starting with 2.0, and the changes
+will automatically be forward-ported to higher versions.
+
+## Restrictions of the development/X.Y branch
+
+The `development/X.Y` branch is a very specific branch in our workflow. No 
+commit must ever go directly into the master branch, and the code shall follow
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
commit must ever go directly into any of these branches, and the code shall follow
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+one and only one path to get into the master branch.
```
Comment (NicolasT) [inline CONTRIBUTING.md]:
```suggestion
one and only one path to get into the per-version release branches.
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+## Workflow Guidelines
+
+### Branching Guidelines
+
+When branching, one should choose a name:
+
+* containing a reference to the GitHub issue being tackled
+* starting with a letter, due to a restriction in our merging bot
+* relevant to the goal of the changes made
+
```
Comment (NicolasT) [inline CONTRIBUTING.md:44]: Mention the valid branch-name prefixes? That can then replace the 2 bullet points above.

Also, maybe provide an example?

**Review by NicolasT — CHANGES_REQUESTED** [review]
Some minor requests.

File: CONTRIBUTING.md (hunk 6 above)
Comment (jbertran) [reply CONTRIBUTING.md]: Removed all occurrences.

**Review by NicolasT — APPROVED** [review]

**Comment by jbertran** [comment]
All comments taken into account in PR with new branch name: #702 .
Declining this one due to branch name because I'm dumb.

**Pull request closed without merging**